### PR TITLE
Move superscript asterisks to subscript

### LINKED ISSUE #18: Double superscript
The asterisk plus prime is very hard to read.

`* Let ''R<sup>*</sup>' = R<sup>*</sup><sub>1</sub> + b⋅R<sup>*</sup><sub>2</sub>''`

https://github.com/jonasnick/bips/blame/aab260c37112b2010266fd8f0d3d56ea1111a2cc/bip-musig2.mediawiki#L382

Sorry for opening an entire issue for this but I don't have a great suggestion right now

### PR DESCRIPTION
**Pull request by real-or-random**
Except for pk, where we just drop them because there's no risk of confusing pk
with aggpk.

Fixes #18.

### CODE REVIEW

**Comment by real-or-random** [comment]
Okay, LOW ASTERISK looks good enough actually... Not entirely sure if it will be too brittle with other fonts but I think it's fine. We have other also other non-ASCII symbols, e.g., the middots.

File: bip-musig2.mediawiki [hunk 1]
```diff
@@ -335,8 +335,8 @@ Input:
 ** Let ''extra_in = empty_bytestring''
 * Let ''k<sub>i</sub> = int(hash<sub>MuSig/nonce</sub>(rand || bytes(1, len(aggpk)) || aggpk || m_prefixed || bytes(4, len(extra_in)) || extra_in || bytes(1, i - 1))) mod n'' for ''i = 1,2''
 * Fail if ''k<sub>1</sub> = 0'' or ''k<sub>2</sub> = 0''
-* Let ''R<sup>*</sup><sub>1</sub> = k<sub>1</sub>⋅G, R<sup>*</sup><sub>2</sub> = k<sub>2</sub>⋅G''
-* Let ''pubnonce = cbytes(R<sup>*</sup><sub>1</sub>) || cbytes(R<sup>*</sup><sub>2</sub>)''
+* Let ''R<sub>1,⁎</sub> = k<sub>1</sub>⋅G, R<sub>2,⁎</sub> = k<sub>2</sub>⋅G''
```
Comment (jonasnick) [inline bip-musig2.mediawiki:338]: Shouldn't it be `R<sub>⁎,1</sub>` to be consistent with nonc agg (where signer index is first)?

**Review by jonasnick — COMMENTED** [review]
note: this probably should have a changelog entry

File: bip-musig2.mediawiki [hunk 2]
```diff
@@ -430,17 +430,17 @@ Input:
 * Run ''PartialSigVerifyInternal(psig, pubnonce<sub>i</sub>, pk<sub>i</sub>, session_ctx)''
 * Return success iff no failure occurred before reaching this point.
 
-'''''PartialSigVerifyInternal(psig, pubnonce, pk<sup>*</sup>, session_ctx)''''':
+'''''PartialSigVerifyInternal(psig, pubnonce, pk, session_ctx)''''':
```
Comment (jonasnick) [inline bip-musig2.mediawiki]: There are still occurences of `pk<sup>*</sup>` in the remarks section.

File: bip-musig2.mediawiki (hunk 1 above)
Comment (real-or-random) [reply bip-musig2.mediawiki:338]: Oh yes, the entire point here was to be consistent with nonce agg...

File: bip-musig2.mediawiki [hunk 3]
```diff
@@ -430,17 +430,17 @@ Input:
...
 * Let ''(Q, gacc, _, b, R, e) = GetSessionValues(session_ctx)''; fail if that fails
 * Let ''s = int(psig)''; fail if ''s &ge; n''
-* Let ''R<sup>*</sup><sub>1</sub> = cpoint(pubnonce[0:33]), R<sup>*</sup><sub>2</sub> = cpoint(pubnonce[33:66])''
-* Let ''R<sup>*</sup>' = R<sup>*</sup><sub>1</sub> + b⋅R<sup>*</sup><sub>2</sub>''
-* Let ''R<sup>*</sup> = R<sup>*</sup>' '' if ''has_even_y(R)'', otherwise let ''R<sup>*</sup> = -R<sup>*</sup>' ''
+* Let ''R<sub>1,⁎</sub> = cpoint(pubnonce[0:33]), R<sub>2,⁎</sub> = cpoint(pubnonce[33:66])''
+* Let ''R<sub>⁎</sub>' = R<sub>1,∗</sub> + b⋅R<sub>2,⁎</sub>''
```
Comment (robot-dreams) [inline bip-musig2.mediawiki]:
```suggestion
* Let ''R<sub>⁎</sub>' = R<sub>1,⁎</sub> + b⋅R<sub>2,⁎</sub>''
```

**Review by robot-dreams — COMMENTED** [review]
Concept ACK.

I was originally expecting that you would change the `'` superscripts but changing the `*` superscripts also makes sense (and is arguably cleaner)!

File: bip-musig2.mediawiki (hunk 1 above)
Comment (real-or-random) [reply bip-musig2.mediawiki:338]: I think it's correct. The signer index is in fact second in NonceAgg.

File: bip-musig2.mediawiki (hunk 3 above)
Comment (real-or-random) [reply bip-musig2.mediawiki]: fixed

File: bip-musig2.mediawiki (hunk 2 above)
Comment (real-or-random) [reply bip-musig2.mediawiki]: fixed

**Comment by real-or-random** [comment]
from #18:

> One can see that in https://github.com/jonasnick/bips/pull/46 "aggregate nonce j" clashes with "this signer's effective nonce". How about we call the latter `Re_*`?.

I took a different approach and changed the local name of the final nonce in PartialSigVerifyInternal. 

Con: Now the final nonce has an inconsistent name in that function.
Pro: I prefer that approach slightly because renaming the effective nonce means that a partial sig is `(Re_*, s)` which feels inconsistent, too. If we have a special notation `Re_*` for "partial nonce", shouldn't we also have a special symbol for "partial s"? (And if the answer is yes, then this would complicate the signing algorithm further etc.)

File: bip-musig2.mediawiki (hunk 1 above)
Comment (jonasnick) [reply bip-musig2.mediawiki:338]: ??? no idea what I was looking at

Comment (jonasnick) [reply bip-musig2.mediawiki:338]: sorry

**Comment by jonasnick** [comment]
I'd prefer the approach to give the effective R of a signer a distinct name. This provides more clarity when talking about these values (A: "bla bla R value", B: "which R value?"). Good point that this reasoning also applies to `s`. In the paper we call it `s_i` or `s_1` for the signer. I don't think this reasoning needs to be applied to other variables in the algorithm necessarily because only partial s and effective R of the signer are "global" variables.

**Comment by jonasnick** [comment]
Superseded by #64

**Pull request closed without merging**